gsw_oceanographic_toolbox.c

```diff
@@ -180,6 +180,39 @@ gsw_add_mean(double *data_in, double *data_out)
 }
 /*
 !==========================================================================
+function gsw_infunnel(sa,ct,p)
+!==========================================================================
+
+! "oceanographic funnel" check for the 75-term equation
+!
+! sa     : Absolute Salinity                                 [g/kg]
+! ct     : Conservative Temperature                          [deg C]
+! p      : sea pressure                                      [dbar]
+!
+! gsw_infunnel : 0, if SA, CT and p are outside the "funnel"
+!                1, if SA, CT and p are inside the "funnel"
+!
+!  Note. The term "funnel" (McDougall et al., 2003) describes the range of
+!    SA, CT and p over which the error in the fit of the computationally
+!    efficient 75-term expression for specific volume in terms of SA, CT
+!    and p was calculated (Roquet et al., 2015).
+*/
+int
+gsw_infunnel(double sa, double ct, double p)
+{
+    return !(p > 8000 ||
+        sa < 0 ||
+        sa > 42 ||
+        (p < 500 && ct < gsw_ct_freezing(sa, p, 0)) ||
+        (p >= 500 && p < 6500 && sa < p * 5e-3 - 2.5) ||
+        (p > 500 && p < 6500 && ct > (31.66666666666667 - p * 3.333333333333334e-3)) ||
```

Comment:
This looks like a trivial bug in the Matlab version.  From the pattern here, which matches the Matlab, it is obvious that the first condition should be "p >= 500", not "p > 500".

Reply:
Fixed!